necsim/impls/no-std/src/cogs/habitat/in_memory.rs

```diff
@@ -30,8 +28,8 @@ pub struct InMemoryHabitat<M: MathsCore> {
 impl<M: MathsCore> Backup for InMemoryHabitat<M> {
     unsafe fn backup_unchecked(&self) -> Self {
         Self {
-            habitat: Final::new(self.habitat.clone()),
-            u64_injection: Final::new(self.u64_injection.clone()),
+            habitat: self.habitat.clone(),
```

Comment:
Implement Clone instead and check for other manual impulse that should just forward to Clone